buildSrc/src/main/groovy/io.seqera.groovy-common-conventions.gradle

```diff
@@ -23,8 +23,8 @@ compileJava {
 }
 
 tasks.withType(GroovyCompile) {
-    sourceCompatibility = '11'
-    targetCompatibility = '11'
+    sourceCompatibility = '17'
+    targetCompatibility = '17'
 }
```

Comment:
This may be a problem. This is used also by nextflow that's still uses Java 11 and Groovy 3

Reply:
This is one of the requirements for Micronaut 4 
so we need to start with Nextflow upgrade

Reply:
I will look into it

Reply:
If i set both java 11 and 17 as compatibility, i get this error:
```
             - Incompatible because this component declares an API of a component compatible with Java 17 and the consumer needed a runtime of a component compatible with Java 11
              - Other compatible attribute:
                  - Doesn't say anything about org.gradle.plugin.api-version (required '7.6.1')
          - Variant 'javadocElements' capability io.micronaut.gradle:micronaut-minimal-plugin:4.1.0 declares a runtime of a component, and its dependencies declared externally:
              - Incompatible because this component declares documentation and the consumer needed a library
              - Other compatible attributes:
                  - Doesn't say anything about its target Java version (required compatibility with Java 11)
                  - Doesn't say anything about its elements (required them packaged as a jar)
                  - Doesn't say anything about org.gradle.plugin.api-version (required '7.6.1')
          - Variant 'runtimeElements' capability io.micronaut.gradle:micronaut-minimal-plugin:4.1.0 declares a runtime of a library, packaged as a jar, and its dependencies declared externally:
              - Incompatible because this component declares a component compatible with Java 17 and the consumer needed a component compatible with Java 11
              - Other compatible attribute:
                  - Doesn't say anything about org.gradle.plugin.api-version (required '7.6.1')
          - Variant 'sourcesElements' capability io.micronaut.gradle:micronaut-minimal-plugin:4.1.0 declares a runtime of a component, and its dependencies declared externally:
              - Incompatible because this component declares documentation and the consumer needed a library
              - Other compatible attributes:
                  - Doesn't say anything about its target Java version (required compatibility with Java 11)
                  - Doesn't say anything about its elements (required them packaged as a jar)
                  - Doesn't say anything about org.gradle.plugin.api-version (required '7.6.1')
          - Variant 'testFixturesApiElements' capability io.micronaut.gradle:micronaut-minimal-plugin-test-fixtures:4.1.0 declares a library, packaged as a jar, and its dependencies declared externally:
              - Incompatible because this component declares an API of a component compatible with Java 17 and the consumer needed a runtime of a component compatible with Java 11
              - Other compatible attribute:
                  - Doesn't say anything about org.gradle.plugin.api-version (required '7.6.1')
          - Variant 'testFixturesRuntimeElements' capability io.micronaut.gradle:micronaut-minimal-plugin-test-fixtures:4.1.0 declares a runtime of a library, packaged as a jar, and its dependencies declared externally:
              - Incompatible because this component declares a component compatible with Java 17 and the consumer needed a component compatible with Java 11
              - Other compatible attribute:
                  - Doesn't say anything about org.gradle.plugin.api-version (required '7.6.1')
```

Reply:
for more information check this run 
https://github.com/seqeralabs/libseqera/actions/runs/9260134594/job/25473348650

Reply:
I discussed with @pgeadas about this yesterday. in Platform they didn't face this kind of issue and Micronaut 4 requires Java 17 as base line 
https://micronaut.io/2023/02/16/micronaut-framework-4-0-with-java-17-baseline/